tests/apps/core/test_templatetags_rfc_5646_locale.py

```diff
@@ -0,0 +1,51 @@
+"""
+Unit tests for the template tags of the search app.
+"""
+from django.core.exceptions import ImproperlyConfigured
+from django.test import TestCase
+from django.test.utils import override_settings
+
+from richie.apps.core.templatetags.rfc_5646_locale import rfc_5646_locale
+
+
+class ReactLocaleSearchTemplateTagsTestCase(TestCase):
+    """
+    Unit test suite to validate the behavior of the `rfc_5646_locale` template filter.
+    """
+
+    @override_settings(RFC_5646_LOCALES=["en-US", "fr-CA", "fr-FR", "es-ES"])
+    def test_templatetags_search_tags_rfc_5646_locale_canada_first(self):
+        """
+        Languages should return the first matching BCP47/RFC5646 locale (Canada comes
+        before France).
+        """
+        # Full locales incl. region
+        self.assertEqual(rfc_5646_locale("fr_FR"), "fr-FR")
+        self.assertEqual(rfc_5646_locale("fr_CA"), "fr-CA")
```

Comment:
as pointed above, full languages in Django will normally be in the form: "en-us" so make sure you also support it.